rs/tests/consensus/utils/src/node.rs

```diff
@@ -47,3 +53,58 @@ pub fn get_node_certified_height(node: &IcNodeSnapshot, log: Logger) -> Height {
     )
     .expect("Should be able to retrieve the certified height")
 }
+
+/// Assert that the given node has deleted its state within the next 5 minutes.
+pub fn assert_node_is_unassigned(node: &IcNodeSnapshot, logger: &Logger) {
+    info!(
+        logger,
+        "Asserting that node {} has deleted its state.",
+        node.get_ip_addr()
+    );
+    // We need to exclude the page_deltas/ directory, which is not deleted on state deletion.
+    // That is because deleting it would break SELinux assumptions.
+    let check = r#"[ "$(ls -A /var/lib/ic/data/ic_state -I page_deltas)" ] && echo "assigned" || echo "unassigned""#;
+    let s = node
+        .block_on_ssh_session()
+        .expect("Failed to establish SSH session");
+
+    ic_system_test_driver::retry_with_msg!(
+        format!("check if node {} is unassigned", node.node_id),
+        logger.clone(),
+        secs(300),
+        secs(10),
+        || match execute_bash_command(&s, check.to_string()) {
+            Ok(s) if s.trim() == "unassigned" => Ok(()),
+            Ok(s) if s.trim() == "assigned" => {
+                bail!("Node {} is still assigned.", node.get_ip_addr())
+            }
+            Ok(s) => bail!("Received unexpected output: {}", s),
+            Err(e) => bail!("Failed to read directory: {}", e),
+        }
+    )
+    .expect("Failed to detect that node has deleted its state.");
+
+    let state_removal_failed = "orchestrator_state_removal_failed_total".to_string();
+    let fs_trim_duration = "orchestrator_fstrim_duration_milliseconds".to_string();
+    let fetcher = MetricsFetcher::new_with_port(
+        std::iter::once(node.clone()),
+        vec![state_removal_failed.clone(), fs_trim_duration.clone()],
+        9091,
```

Comment:
Why does port 9091 work here? Where is the port for the metrics defined?

Reply:
Previously it used 9090 by default, which is the port for the replica metrics. But since we want to fetch orchestrator metrics we have to use 9091 here (I'm not sure where that is defined).